app/server.go

```diff
@@ -1581,27 +1580,36 @@ func (s *Server) InitMetricsRouter() error {
 }
 
 func (s *Server) startMetricsServer() {
+	var notify chan struct{}
 	s.metricsLock.Lock()
-	defer s.metricsLock.Unlock()
+	defer func() {
+		if notify != nil {
+			<-notify
+		}
+		s.metricsLock.Unlock()
+	}()
 
-	if s.metricsServer != nil {
+	l, err := net.Listen("tcp", *s.Config().MetricsSettings.ListenAddress)
+	if err != nil {
+		mlog.Error(err.Error())
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.

Reply:
Wow, thanks for reminding this to me @mattermod 🙏